censustree/arbotree/wrapper.go

```diff
@@ -0,0 +1,206 @@
+// Package arbotree provides the functions for creating and managing an arbo
+// merkletree adapted to the CensusTree interface
+package arbotree
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/vocdoni/arbo"
+	"go.vocdoni.io/dvote/censustree"
+	"go.vocdoni.io/dvote/db"
+)
+
+// Tree implements the censustree.Tree interface using arbo.Tree
+// functionallities
+type Tree struct {
+	Tree           *arbo.Tree
+	public         uint32
+	lastAccessUnix int64 // a unix timestamp, used via sync/atomic
+}
+
+// check that censustree.Tree interface is matched by Tree
+var _ censustree.Tree = (*Tree)(nil)
+
+// NewTree opens or creates a merkle tree using the given storage, and the given
+// hash function.
+// Note that each tree should use an entirely separate namespace for its database keys.
+func NewTree(name, storageDir string, hashFunc arbo.HashFunction) (censustree.Tree, error) {
+	dbDir := filepath.Join(storageDir, "arbotree.db."+strings.TrimSpace(name))
+	database, err := db.NewBadgerDB(dbDir)
+	if err != nil {
+		return nil, err
+	}
+
+	mt, err := arbo.NewTree(database, 140, hashFunc)
+	if err != nil {
+		return nil, err
+	}
+	tree := &Tree{
+		Tree: mt,
+	}
+	tree.updateAccessTime()
+	return tree, nil
+}
+
+func (t *Tree) Init(name, storageDir string) error {
+	dbDir := filepath.Join(storageDir, "arbotree.db."+strings.TrimSpace(name))
+	database, err := db.NewBadgerDB(dbDir)
+	if err != nil {
+		return err
+	}
+
+	// by default is used the Blake2b hash function. If another hash
+	// function is desired, should use NewTree method or even
+	// censustree/util.NewCensusTree method, which allows to choose the hash
+	// function to be used in the Tree.
+	mt, err := arbo.NewTree(database, 140, arbo.HashFunctionBlake2b)
+	if err != nil {
+		return err
+	}
+	t.Tree = mt
+	t.updateAccessTime()
+	return nil
+}
+
+func (t *Tree) MaxKeySize() int {
```

Comment:
Add description to all Exported methods please.

Reply:
Done, also added the descriptions to the methods of `censustree` & `graviton` packages.